vm/systemSmartContracts/governance.go

```diff
@@ -1082,6 +1073,31 @@ func (g *governanceContract) getUserVotes(address []byte) (*OngoingVotedListV2,
 	return onGoingList, nil
 }
 
+func (g *governanceContract) getUserVotesV2(address []byte) (*OngoingVotedListV2, error) {
+	onGoingListV2 := &OngoingVotedListV2{
+		Direct:               make([]uint64, 0),
+		DelegatedWithAddress: make([]*DelegatedWithAddress, 0),
+	}
+	marshaledData := g.eei.GetStorage(address)
+	if len(marshaledData) == 0 {
+		return onGoingListV2, nil
+	}
+
+	err := g.marshalizer.Unmarshal(onGoingListV2, marshaledData)
+	if err == nil {
+		return onGoingListV2, nil
+	}
+	onGoingList := &OngoingVotedList{
+		Direct:    make([]uint64, 0),
+		Delegated: make([]uint64, 0),
+	}
+	err = g.marshalizer.Unmarshal(onGoingList, marshaledData)
+	if err != nil {
+		return nil, err
+	}
+	return onGoingListV2, nil
```

Comment:
You should return onGoingList not the v2

Reply:
+1

Reply:
this method will be called only after GovernanceFixes gets activated, that means there is no ongoing v1 proposal, thus the information is not necessary anymore.